### PR TITLE
Save metadata during "coda build"

### PR DESCRIPTION
**Pull request by dweitzman-codaio**
I don't think there are any other commands that can print the metadata, but it can be helpful for debugging.

It's marked as "hidden" since it seems useful to the packs team but not necessarily for pack makers.

### CODE REVIEW

**Comment by jonathan-codaio** [comment]
Cool, this is definitely super useful.

Thoughts on doing this as part of `coda build`, since that already outputs compiled output to a temp directory? Should we just add the metadata to the temporary output?

**Comment by dweitzman-codaio** [comment]
> Cool, this is definitely super useful.
> 
> Thoughts on doing this as part of `coda build`, since that already outputs compiled output to a temp directory? Should we just add the metadata to the temporary output?

Makes sense to me. Updated the PR

File: testing/compile.ts [hunk 1]
```diff
@@ -271,6 +273,12 @@ export async function compilePackBundle({
     throw await processVmError(err, tempBundlePath);
   }
 
+  // Write the generated metadata. It's not used by the upload command, but
+  // it's helpful for debugging upload validation errors.
+  const metadata = await importManifest<PackVersionDefinition>(tempBundlePath);
+  const tempMetadataPath = path.join(intermediateOutputDirectory, 'metadata.json');
+  fs.writeFileSync(tempMetadataPath, JSON.stringify(metadata));
```
Comment (jonathan-codaio) [inline testing/compile.ts:280]: This should be writing the result of `compilePackMetadata`, right?

Comment (dweitzman-codaio) [reply testing/compile.ts:282]: Good catch, fixed

**Review by jonathan-codaio — APPROVED** [review]

**Review by huayang-codaio — COMMENTED** [review]
can be very useful. thank you

**Pull request merged**